sigal/gallery.py

```diff
@@ -479,7 +489,11 @@ def zip(self):
         zip_gallery = self.settings['zip_gallery']
 
         if zip_gallery and len(self) > 0:
+            zip_gallery = zip_gallery.format(album=self)
             archive_path = join(self.dst_path, zip_gallery)
+            if isfile(archive_path):
+                self.logger.debug("Archive %s already created, passing", archive_path)
+                return zip_gallery
```

Comment:
I agree on the issue, but this case is more problematic. Without a way to know of images have been added or removed, you cannot tell if the file must be regenerated.
But there may be at least a solution to avoid generating this file multiple times. Using the `@cached_property` decorator should work.